### PR TITLE
Add ESLint & Prettier configurations

### LINKED ISSUE #27: Add ESLint & Prettier support to keep a check on development standard practices.
The general public has general opinion on the best development practices all the time. Some prefers using double quotes over single quotes, while others prefer semi-colons after a line & others don't. Debates on such meaningless topics are unproductive, as such having a proper [ESLint](https://eslint.org) & [Prettier](https://prettier.io) configuration in place should keep a check on a standard development practice.

For reference on how to setup the configurations for each of those dependencies, refer to ["Configuring ESLint"](https://eslint.org/docs/user-guide/configuring/) & ["Prettier Configuration Options"](https://prettier.io/docs/en/options.html)

### PR DESCRIPTION
**Pull request by afkcodes**
## Description

This PR configures ESLint and Prettier to the project enables better code standards and parity.
Airbnb Config has been used.

Fixes #27 

## Changes Include

- [x] Bug fix 🐛 (non-breaking change which fixes an issue)
- [x] New feature ✨ (non-breaking change which adds functionality)
- [x] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)

## How Has This Been Tested?

- [x] Tested incorrect formatting with eslint warnings.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by afkcodes** [comment]
I really don't know why the check is failing seems like a peer-dep issue:

**Comment by Jarmos-san** [comment]
@afkcodes I skimmed through the contents of this PR & here's a couple of things I would like to mention:

1. Next.js v11.x supports [ESLint out-of-the-box with amazing support](https://nextjs.org/docs/basic-features/eslint)!
2. The oob config Next.js provides is good enough for the project, as such our config needn't have anything other than the following contents:
    ```json
    {
        "extends": ["next/core-web-vitals", "prettier"]
    }
    ```
3. The configurations for both ESLint & Prettier this PR provides is a bit too opinionated for the project's requirements (_we would prefer resorting to standard dev practices wherever possible_).
4. Adding more dependencies is subject to discussion since it expands the potential area of stumbling across a breaking bug (we've been there with this project itself just a while ago). So, try not to include any other ESLint/Prettier plugins other than the Next.js recommended ones.

I'll perform a more thorough review & will work with you to polish this PR further by this evening. But for starters, you can please ensure you've `.gitignore`-ed the `.vscode` directory (not everyone uses VSCode). And simplify the ESLint configurations as well after referring to the [Next.js + ESLint docs](https://nextjs.org/docs/basic-features/eslint). I bet the custom ESLint this PR provides is conflicting with the `next lint` wrapper in the CI.

**Comment by afkcodes** [comment]
seems like so, i will update, the PR

**Comment by afkcodes** [comment]
I have updated it in the local and seems like the prettier is not kicking off.  Doesn't show any issues with the wrong formatting.

**Review by Jarmos-san — CHANGES_REQUESTED** [review]
PR looks pretty nice but it needs to be a bit more minimal so as to keep room for future improvements/additions to the project. Please refer to the individual comments & make the changes which should also fix the CI/CD pipeline as well.

File: package.json [hunk 1]
```diff
@@ -11,11 +11,26 @@
   },
   "dependencies": {
     "next": "12.0.10",
+    "npm": "^8.4.1",
```
Comment (Jarmos-san) [inline package.json:14]: Could you remove this as a dependency? We really don't want to expand the potential area of vulnerability if a dependency breaks for whatever reason. We assume every contributor already has Node.js installed which already packages `npm` with it. So, there's pretty much no reason to include this entry in the `package.json` file.

File: package.json [hunk 2]
```diff
@@ -11,11 +11,26 @@
...
     "react": "17.0.2",
     "react-dom": "17.0.2"
   },
   "devDependencies": {
     "@types/react": "^17.0.39",
+    "@typescript-eslint/eslint-plugin": "^5.10.2",
```
Comment (Jarmos-san) [inline package.json:20]: As mentioned earlier, Next.js v11.x comes pre-packaged with amazing support for ESLint out-of-the-box! We really don't need the additional dev-dependencies other than: `eslint`, `eslint-config-next` & `prettier` (although we might require some Prettier plugins). So, could you remove the unnecessary dependencies?

That said, here's the change you need to make to the `package.json` file:

```diff
  "devDependencies": {
    "@types/react": "^17.0.39",
-   "@typescript-eslint/eslint-plugin": "^5.10.2",
-   "@typescript-eslint/parser": "^5.10.2",
    "eslint": "^7.23.0",
-   "eslint-config-airbnb": "^19.0.4",
    "eslint-config-next": "12.0.2",
-   "eslint-config-prettier": "^8.3.0",
-   "eslint-import-resolver-typescript": "^2.5.0",
-   "eslint-plugin-import": "^2.25.4",
-   "eslint-plugin-jsx-a11y": "^6.5.1",
-   "eslint-plugin-prettier": "^4.0.0",
-   "eslint-plugin-react": "^7.28.0",
-   "eslint-plugin-react-hooks": "^4.3.0",
-   "@next/eslint-plugin-next": "^12.0.10",
    "prettier": "^2.4.1",
    "typescript": "^4.5.5"
  }
```

**Review by Jarmos-san — CHANGES_REQUESTED** [review]
Forget to mention about adding the ESLint plugins as dev-dependencies which aren't exactly necessary since we're using Next.js v12.x (and Next.js v11.x supported ESLint out-of-the-box **really** well!)

File: package.json (hunk 1 above)
Comment (afkcodes) [reply package.json:14]: ya this got introduced i have removed it

File: package.json (hunk 2 above)
Comment (afkcodes) [reply package.json:20]: why do we need eslint if it comes by default with next?

**Comment by afkcodes** [comment]
Updated the PR, but it seems the auto correct in my VScode still didn't kicked into fix issues.

File: package.json (hunk 2 above)
Comment (Jarmos-san) [reply package.json]: Next.js provides some "default configurations" for ESLint which gets depending on the contents of the `.eslintrc.json`. I really suggest taking a look at the [official docs](https://nextjs.org/docs/basic-features/eslint) on how Next.js wraps ESLint to provide a sensible out-of-the-box experience.

You still need to add the `eslint` binary as a dependency, else `npm lint` won't work as expected. The `eslint-config-next` wraps the other ESLint configs & plugins like `eslint-plugin-jsx-a11y`, `eslint-plugin-react` & so on. Please take a look at the docs for more info on it.

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "bracketSameLine": false
```
Comment (Jarmos-san) [inline .prettierrc:2]:
```diff
{
-   "bracketSameLine": false
+   "bracketSameLine": true
}
```

The `"bracketSameLine": false` is the default behaviour, by having the ending bracket on the same line makes the code look clean.

**Comment by afkcodes** [comment]
Updated the PR and the check still fails, have a look if i missed something.

**Review by Jarmos-san — CHANGES_REQUESTED** [review]
PR is in a much better position right now, but seems like the contents of the directory `.vscode` is still cached in VCS. See if you could remove them by invoking the following command:

```shell
git rm --cached ".vscode"
```

File: .prettierrc (hunk 3 above)
Comment (afkcodes) [reply .prettierrc:2]: updating

**Comment by afkcodes** [comment]
updated, i still don't know why the checks are failing. lol

**Review by Jarmos-san — APPROVED** [review]
Contents of the PR looks good enough but for whatever reason the status check fails. Since, it's difficult to take a look at the logs in a CI/CD environment, I'm merging this PR for now & will keep a track on it for possible breakages in the future.

**Comment by afkcodes** [comment]
Oh man, the config was right the whole time :D well, i am still not so aligned with the config we have, what i generally use in my projects keeps things streamlined with the eslint and prettier config.

**Pull request merged**